### PR TITLE
[tools] Add lint tool for catching common mistakes in test files

### PR DESCRIPTION
**Pull request by yunxliu**
- The lint tool can catch common mistakes in test files. Now it can check html, htm, xhtml, xhtm, json and xml files.

https://crosswalk-project.org/jira/browse/CTS-258

### CODE REVIEW

**Comment by Honry** [comment]
Miss travis.yml file

**Comment by Honry** [comment]
Have you tried on Travis CI?

File: tools/lint/sourcefile.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
+import os
```
Comment (Honry) [inline tools/lint/sourcefile.py]: Please add license file.

File: tools/lint/utils.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import os
+import urlparse
+from StringIO import StringIO
+
+blacklist = ["/", "/tools/", "/resources/", "/common/", "/conformance-checkers/", "_certs"]
```
Comment (haoxli) [inline tools/lint/utils.py]: Not used. please remove it.

File: tools/lint/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+* **EARLY-TESTHARNESSREPORT**: Test file has an instance of
+  `<script src='/resources/testharnessreport.js'>` prior to
+  `<script src='/resources/testharness.js'>`; **fix**: flip the order.
+
+* **INDENT TABS**: Test-file line starts with one or more tab characters;
+  **fix**: use spaces to replace any tab characters at beginning of lines.
+
+* **MISSING-TESTHARNESSREPORT**: Test file is missing an instance of
+  `<script src='/resources/testharnessreport.js'>`; **fix**: ensure each
+  test file contains `<script src='/resources/testharnessreport.js'>`.
```
Comment (haoxli) [inline tools/lint/README.md]: Test file is missing an instance of  `<script src='/resources/testharnessreport.js'>` if there is testharness.js.
Also add **MISSING-TESTHARNESS**.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+language: python
+
+python:
+  - "2.7"
+
+before_install:
+  - sudo apt-get install pylint
+  - sudo apt-get install python-demjson
+  - sudo apt-get install libxml2-utils
+  - git submodule update --init --recursive
```
Comment (Honry) [inline .travis.yml]: There's no submodule in crosswalk-test-suite.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+script: python ./tools/lint/lint.py
```
Comment (Honry) [inline .travis.yml]: Only related files in PR need check when testing with travis.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+language: python
+
+python:
+  - "2.7"
+
+before_script:
+  - sudo apt-get install pylint
+  - sudo apt-get install python-demjson
+  - sudo apt-get install libxml2-utils
+  - sudo pip install html5lib
```
Comment (Honry) [inline .travis.yml]: Is html5lib module necessary?

File: tools/lint/README.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  `<script src='/resources/testharnessreport.js'>`; **fix**: ensure each test
+  has only one `<script src='/resources/testharnessreport.js'>` instance.
+
+* **PATH LENGTH**: Test file's pathname has a total length greater than 150
+  characters; **fix**: use shorter filename to rename the test file.
+
+* **TRAILING WHITESPACE**: Test-file line has trailing whitespace; **fix**:
+  remove trailing whitespace from all lines in the file.
+
+* **EXTERNAL EXECUTABLE PERMISSION**: Test file contains external executable permission; **fix**:
```
Comment (Honry) [inline tools/lint/README.md]: What does external mean?

File: tools/lint/item.py [hunk 8]
```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env python
```
Comment (Honry) [inline tools/lint/item.py]: This tool is integrated from W3C, the license should not be Intel, please create a COPYING file to record the license.

File: tools/lint/README.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
+## Introduction
+
+We have a lint tool for catching common mistakes in test files. Now it can
+check html, htm, xhtml, xhtm, json and xml files.
+- You can run it manually by starting the `lint.py` executable from the root of your local
+crosswalk-test-suite working directory like this:
+
+```
+./tools/lint/lint.py
```
Comment (Honry) [inline tools/lint/README.md]: You'd better add "-f" here, as a separated section to introduce checking all test suite.

File: tools/lint/COPYING [hunk 10]
```diff
@@ -0,0 +1,38 @@
+This test suite comes from
+https://github.com/w3c/wpt-tools/
+with modification:
+
+lint.py
+
+ -def all_git_paths():
```
Comment (Honry) [inline tools/lint/COPYING]: No need to detail the modification, just summary your update with brief sentence.

File: tools/lint/README.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+```
+
+The lint tool is also run automatically for every submitted pull request,
+and reviewers will not merge branches with tests that have lint errors, so
+you must either [fix all lint errors](#fixing-lint-errors), or you must
+[white-list test files] (#updating-the-whitelist) to suppress the errors.
+(Note: If check all crosswalk-test-suite, need add `-f` behind `python ./tools/lint/lint.py` in .travis.yml file, `script: python ./tools/lint/lint.py -f`. If check
```
Comment (Honry) [inline tools/lint/README.md]: A new section for introduce the usage of lint.py would be pretty good.

File: .travis.yml [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+python:
+  - "2.7"
+
+before_script:
+  - sudo apt-get install pylint
+  - sudo apt-get install python-demjson
+  - sudo apt-get install libxml2-utils
+  - pip install html5lib
+
+script: python ./tools/lint/lint.py -f
```
Comment (Honry) [inline .travis.yml]: Please remove '-f', we only need to check PR.

File: tools/lint/README.md [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+
+* **MULTIPLE-TESTHARNESS**: Test file with multiple instances of
+  `<script src='/resources/testharness.js'>`; **fix**: ensure each test
+  has only one `<script src='/resources/testharness.js'>` instance.
+
+* **MULTIPLE-TESTHARNESSREPORT**: Test file with multiple instances of
+  `<script src='/resources/testharnessreport.js'>`; **fix**: ensure each test
+  has only one `<script src='/resources/testharnessreport.js'>` instance.
+
+* **PATH LENGTH**: Test file's pathname has a total length greater than 150
```
Comment (Honry) [inline tools/lint/README.md]: This error is not suitable in Crosswalk test suite.

File: tools/lint/README.md [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+
+```
+TRAILING WHITESPACE:example/file.html:128
+```
+
+## Authors:
+
+* Yun, Liu<yunx.liu@intel.com>
+
+## LICENSE
```
Comment (Honry) [inline tools/lint/README.md]: The copyright should be W3C.

File: tools/lint/README.md [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+</code>
+1. If check specified folder, need add `-d` behind `python ./tools/lint/lint.py`
+in .travis.yml file:</br>
+<code>
+script: python ./tools/lint/lint.py -d apptools
+</code>
+
+## Setup Environment
+
+1. Install pylint: `sudo apt-get install pylint`
```
Comment (Honry) [inline tools/lint/README.md]: Where's the error from pylint?

File: tools/lint/README.md [hunk 16]
```diff
@@ -0,0 +1,135 @@
+## Introduction
+
+We have a lint tool for catching common mistakes in test files. Now it can
+check html, htm, xhtml, xhtm, json and xml files.
```
Comment (Honry) [inline tools/lint/README.md]: miss python file.

File: tools/lint/lint.py [hunk 17]
```diff
@@ -0,0 +1,275 @@
+# !/user/bin/env python
```
Comment (Honry) [inline tools/lint/lint.py]: #! /usr/bin/env python

**Comment by Honry** [comment]
LGTM

**Pull request merged**